test/spec/features/service/link_spec.rb

```diff
@@ -1,44 +1,40 @@
 describe 'service link' do
   after(:each) do
-    %w(test-1 test-2).each do |s|
-      run "kontena service rm --force #{s}"
-    end
+    run "kontena service rm --force test-1"
+    run "kontena service rm --force test-2"
     run "kontena stack rm --force simple"
   end
 
   it 'links service to target' do
     run "kontena service create test-1 redis:3.0"
     run "kontena service create test-2 redis:3.0"
-    k = run "kontena service link test-1 test-2"
-    expect(k.code).to eq(0)
-    k = run "kontena service show test-1"
+    run! "kontena service link test-1 test-2"
+    k = run! "kontena service show test-1"
     expect(k.out.match(/^\s+- test-2\s*$/)).to be_truthy
   end
 
   it 'links service to stack service' do
     with_fixture_dir("stack/simple") do
-      run 'kontena stack install --no-deploy'
+      run! 'kontena stack install --no-deploy'
     end
-    run "kontena service create test-1 redis:3.0"
-    k = run "kontena service link test-1 simple/redis"
-    expect(k.code).to eq(0)
-    k = run "kontena service show test-1"
+    run! "kontena service create test-1 redis:3.0"
+    run! "kontena service link test-1 simple/redis"
+    k = run! "kontena service show test-1"
     expect(k.out.match(/^\s+\- simple\/redis\s*$/)).to be_truthy
   end
 
   it 'links stack service with existing links to service' do
     with_fixture_dir("stack/links") do
-      run 'kontena stack install --no-deploy links.yml'
+      run! 'kontena stack install --no-deploy links.yml'
     end
-    run "kontena service create test-1 redis:3.0"
-    k = run "kontena service link simple/bar test-1"
-    expect(k.code).to eq(0)
-    k = run "kontena service show simple/bar"
+    run! "kontena service create test-1 redis:3.0"
+    run! "kontena service link simple/bar test-1"
+    k = run! "kontena service show simple/bar"
     expect(k.out.match(/^\s+\- test-1\s*$/)).to be_truthy
   end
 
   it 'returns error if target does not exist' do
-    run "kontena service create test-1 redis:3.0"
+    run! "kontena service create test-1 redis:3.0"
```

Comment:
I wonder if this has actually always been failing before? I don't immediately see why, though...

```
service link
  links service to target
  links service to stack service
  links stack service with existing links to service
  returns error if target does not exist (FAILED - 1)

Failures:

  1) service link returns error if target does not exist
     Failure/Error: raise Error.new(cmd, k.code, k.out) unless k.code.zero?
     
     Shell::Error:
       command failed with code 1: kontena service create test-1 redis:3.0
 [fail] Creating test-1 service      e  .. \
        [error] 422 : Unprocessable Entity:
       	name: is already taken
     # ./spec/support/shell.rb:28:in `block in run!'
     # ./spec/support/shell.rb:27:in `tap'
     # ./spec/support/shell.rb:27:in `run!'
     # ./spec/features/service/link_spec.rb:37:in `block (2 levels) in <top (required)>'
```

Reply:
It looks like the `links stack service with existing links to service` spec is leaking the `test-1` service...

```
$ rspec spec/features/service/link_spec.rb -e 'links stack service with existing links to service'
Run options: include {:full_description=>/links\ stack\ service\ with\ existing\ links\ to\ service/}

service link
  links stack service with existing links to service

Top 1 slowest examples (4.11 seconds, 100.0% of total time):
  service link links stack service with existing links to service
    4.11 seconds ./spec/features/service/link_spec.rb:26

Finished in 4.11 seconds (files took 0.09469 seconds to load)
1 example, 0 failures

$ kontena service list
NAME       INSTANCES   STATEFUL   STATE         EXPOSED PORTS
⊝ test-1   0 / 1       no         initialized
```